vmm/src/vm.rs

```diff
@@ -295,7 +296,7 @@ impl Vm {
     fn new_from_memory_manager(
         config: Arc<Mutex<VmConfig>>,
         memory_manager: Arc<Mutex<MemoryManager>>,
-        vm: Arc<dyn hypervisor::Vm>,
+        vm: Arc<Mutex<dyn hypervisor::Vm>>,
```

Comment:
Secondly, the logic of setting vmmops to None and eventually setting it to correct Object forced me to change the return type of **create_vm** to be mutable (from hypervisor::Result<Arc> -> Result<Arc<Mutex>>). This change seems sensible.

Now the challenge is changing the inputs for vm in other struct like device_manager, cpu_manager, memory_manager. None of these objects need a Mutable Vm object. But the above change is forcing the vm object passed to these objects to be mutable. Any way I can avoid having to pass mutable Vm object to these Objects?

Reply:
I don't think that matters. If there is some contention over the `Mutex` then `RwLock` can peform better (but only when you have multiple regular readers / infrequent writers.